### ISSUE TITLE
Crash (in Python) on an URL from www.msnbc.com

### ISSUE THREAD

**Issue by rsfinlayson** [ytdl-org/youtube-dl#18883]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.01.16*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [ x] I've **verified** and **I assure** that I'm running youtube-dl **2019.01.16**

### Before submitting an *issue* make sure you have:
- [x ] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [ x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'http://www.youtube.com/watch?v=BaW_jenozKcj']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2019.01.16
[debug] Python version 2.7.11 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: ffmpeg N-75573-g1d0487f, ffprobe N-75573-g1d0487f, rtmpdump 2.4
[debug] Proxy map: {}
...
<end of log>
```

---

youtube-dl -v 'https://www.msnbc.com/hardball/watch/bill-maher-trump-should-be-impeached-1426509891917?cid=sm_npd_ms_fb_hb'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://www.msnbc.com/hardball/watch/bill-maher-trump-should-be-impeached-1426509891917?cid=sm_npd_ms_fb_hb']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.01.16
[debug] Python version 2.7.14 (CPython) - Darwin-18.2.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.3.4, ffprobe 3.3.4, rtmpdump 2.4
[debug] Proxy map: {}
[ThePlatformFeed] 1426509891917: Downloading JSON metadata
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/local/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 472, in main
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 462, in _real_main
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2002, in download
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 804, in extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 870, in process_ie_result
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 793, in extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 508, in extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/theplatform.py", line 411, in _real_extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/theplatform.py", line 345, in _extract_feed_info
IndexError: list index out of range
---

### Description of your *issue*, suggested solution and other information

Explanation of your *issue* in arbitrary form goes here. Please make sure the [description is worded well enough to be understood](https://github.com/rg3/youtube-dl#is-the-description-of-the-issue-itself-sufficient). Provide as much context and examples as possible.
If work on your *issue* requires account credentials please provide them or explain how one can obtain them.

youtube-dl crashes when given the following URL (that plays OK in a browser):
https://www.msnbc.com/hardball/watch/bill-maher-trump-should-be-impeached-1426509891917?cid=sm_npd_ms_fb_hb

**Comment by Tanath** [comment]
I'm also getting this on MSNBC.

```
youtube-dl -v 'https://www.msnbc.com/rachel-maddow/watch/barr-scrambles-to-rewind-spying-conspiracy-theory-embrace-1489359427929'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.msnbc.com/rachel-maddow/watch/barr-scrambles-to-rewind-spying-conspiracy-theory-embrace-1489359427929']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.04.24
[debug] Python version 3.7.3 (CPython) - Linux-5.0.0-0.1-MANJARO-x86_64-with-arch-Manjaro-Linux
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, rtmpdump 2.4
[debug] Proxy map: {}
[ThePlatformFeed] 1489359427929: Downloading JSON metadata
Traceback (most recent call last):
  File "/usr/bin/youtube-dl", line 11, in <module>
    load_entry_point('youtube-dl==2019.4.24', 'console_scripts', 'youtube-dl')()
  File "/usr/lib/python3.7/site-packages/youtube_dl/__init__.py", line 474, in main
    _real_main(argv)
  File "/usr/lib/python3.7/site-packages/youtube_dl/__init__.py", line 464, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/lib/python3.7/site-packages/youtube_dl/YoutubeDL.py", line 2008, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/usr/lib/python3.7/site-packages/youtube_dl/YoutubeDL.py", line 807, in extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/lib/python3.7/site-packages/youtube_dl/YoutubeDL.py", line 873, in process_ie_result
    extra_info=extra_info, download=False, process=False)
  File "/usr/lib/python3.7/site-packages/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3.7/site-packages/youtube_dl/extractor/common.py", line 529, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3.7/site-packages/youtube_dl/extractor/theplatform.py", line 411, in _real_extract
    return self._extract_feed_info(provider_id, feed_id, filter_query, video_id)
  File "/usr/lib/python3.7/site-packages/youtube_dl/extractor/theplatform.py", line 345, in _extract_feed_info
    entry = self._download_json(real_url, video_id)['entries'][0]
IndexError: list index out of range
```

**Comment by Dullaz** [comment]
I can confirm this is still an issue with version 2020.09.20

> $ youtube-dl --verbose https://www.msnbc.com/hardball/watch/bill-maher-trump-should-be-impeached-1426509891917?cid=sm_npd_ms_fb_hb
> [debug] System config: []
> [debug] User config: []
> [debug] Custom config: []
> [debug] Command-line args: [u'--verbose', u'https://www.msnbc.com/hardball/watch/bill-maher-trump-should-be-impeached-1426509891917?cid=sm_npd_ms_fb_hb']
> [debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
> [debug] youtube-dl version 2020.09.20
> [debug] Python version 2.7.18rc1 (CPython) - Linux-4.19.128-microsoft-standard-x86_64-with-Ubuntu-20.04-focal
> [debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4
> [debug] Proxy map: {}
> [NBCNews] 1426509891917: Downloading webpage
> ERROR: Unable to extract bootstrap json; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
> Traceback (most recent call last):
>   File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 797, in extract_info
>     ie_result = ie.extract(url)
>   File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 532, in extract
>     ie_result = self._real_extract(url)
>   File "/usr/local/bin/youtube-dl/youtube_dl/extractor/nbc.py", line 398, in _real_extract
>     'bootstrap json'), video_id, js_to_json)
>   File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 1010, in _search_regex
>     raise RegexNotFoundError('Unable to extract %s' % _name)
> RegexNotFoundError: Unable to extract bootstrap json; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

It seems that it's trying to use the "nbc" extractor while the url is for "msnbc", a different platform.

I can imagine this is probably a regex issue but I wouldn't know where to begin with fixing this.

**Comment by remitamine** [comment]
Fixed with the latest version(2020.11.17).
